docs/data-sources/storage_box.md

```diff
@@ -0,0 +1,80 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "hcloud_storage_box Data Source - hcloud"
+subcategory: ""
+description: |-
+  Provides details about a Hetzner Storage Box.
+  See the Storage Boxes API documentation https://docs.hetzner.cloud/reference/hetzner#storage-boxes for more details.
+---
+
+# hcloud_storage_box (Data Source)
+
+Provides details about a Hetzner Storage Box.
+
+See the [Storage Boxes API documentation](https://docs.hetzner.cloud/reference/hetzner#storage-boxes) for more details.
+
+## Example Usage
+
+```terraform
+data "hcloud_storage_box" "by_id" {
+  id = 1333
+}
+
+data "hcloud_storage_box" "by_name" {
+  name = "backups"
+}
+
+data "hcloud_storage_box" "by_label_selector" {
+  with_selector = "env=production"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `id` (Number) ID of the Storage Box.
+- `name` (String) Name of the Storage Box.
+- `snapshot_plan` (Attributes) Details of the active snapshot plan. (see [below for nested schema](#nestedatt--snapshot_plan))
+- `with_selector` (String) Filter results using a [Label Selector](https://docs.hetzner.cloud/reference/hetzner#label-selector).
+
+### Read-Only
+
+- `access_settings` (Attributes) Access settings of the Storage Box. (see [below for nested schema](#nestedatt--access_settings))
+- `delete_protection` (Boolean) Whether delete protection is enabled.
+- `labels` (Map of String) User-defined [labels](https://docs.hetzner.cloud/reference/cloud#labels) (key-value pairs) for the resource.
+- `location` (String) Name of the Location.
+- `server` (String) FQDN of the Storage Box.
+- `storage_box_type` (String) Name of the Storage Box Type.
+- `system` (String) Host system of the Storage Box.
+- `username` (String) Primary username of the Storage Box.
+
+<a id="nestedatt--snapshot_plan"></a>
+### Nested Schema for `snapshot_plan`
+
+Read-Only:
+
+- `day_of_month` (Number) Day of the month when the Snapshot Plan is executed.
+
+Null means every day.
```

Comment:
Should we remove the empty line here? I think the new line is good, but the extra new lines might be too much.